io-engine/src/lvs/lvs_lvol.rs

```diff
@@ -1008,25 +1016,27 @@ impl LvsLvol for Lvol {
         }
     }
 
-    /// Build Snapshot Parameters from Blob.
-    fn build_snapshot_param(&self, _blob: *mut spdk_blob) -> SnapshotParams {
-        // TODO: need to Integrate with Snapshot Property Enumeration
-        // Currently it is stub.
-        SnapshotParams::new(
-            Some(self.name()),
-            Some(self.name()),
-            Some(self.name()),
-            Some(self.name()),
-            Some(self.name()),
-            Some(Utc::now().to_string()),
-            false,
-        )
+    /// Get the parent spdk_blob from the current blob.
+    unsafe fn bs_iter_parent(
+        &self,
+        curr_blob: *mut spdk_blob,
+    ) -> Option<*mut spdk_blob> {
+        unsafe {
+            let parent_blob = spdk_bs_iter_parent(curr_blob);
+            if parent_blob.is_null() {
+                None
+            } else {
+                Some(parent_blob)
+            }
+        }
```

Comment:
Function is already marked as unsafe

```suggestion
        let parent_blob = spdk_bs_iter_parent(curr_blob);
        if parent_blob.is_null() {
            None
        } else {
            Some(parent_blob)
        }
```

Reply:
fixed.